release/models/platform/openconfig-platform-healthz.yang

```diff
@@ -0,0 +1,127 @@
+     further diagnostic and debugging informaton from a network
+     device.";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2023-01-23" {
+    description
+      "Initial healthz variable revision";
+    reference "0.1.0";
+  }
+
+  grouping platform-health-top {
+    description
+      "Grouping containing health-related parameters.";
+
+    container healthz {
+      description
+        "The health of the component. The paramaters within this
+        container indicate the status of the component beyond whether
+        it is operationally up or down. When a signal is received
+        that a component is in an unhealthy state the gNOI.Healthz
+        service can be used to retrieve further diagnostic information
+        relating to the component.
+
+        The contents of this directory relate only to the specific
+        component that it is associated with. In the case that child
+        components become unhealthy and this causes a parent component
+        to be unhealthy, the new unhealthy status should be reported at
+        both components, such that an interested system can take the
+        relevant actions (e.g., retrieve the Healthz output, or
+        apply mitigation actions).";
+      reference
+        "https://github.com/openconfig/gnoi/tree/main/healthz";
+
+      container state {
+        config false;
+        description
+          "Operational state parameters relating to component health.";
+        uses platform-health-state;
+      }
+    }
+  }
+
+  grouping platform-health-state {
+    description
+      "Operational state parameters relating to a platform component's
+      health.";
+
+    leaf status {
+      type enumeration {
+        enum UNSPECIFIED {
+          description
+            "The component's health status has not yet been checked
+            by the system.";
+        }
+
+        enum HEALTHY {
+          description
+            "The component is in a HEALTHY state, and is operating
+            within the expected parameters.";
+        }
+
+        enum UNHEALTHY {
+          description
+            "The component is in a unhealthy state, it is not
+            performing the function expected of it.";
+        }
+      }
+      description
+        "The status of the component, indicating its current health.";
+      oc-ext:telemetry-on-change;
+    }
+
+    leaf last-unhealthy {
+      type oc-types:timeticks64;
+      description
+        "The time at which the component as last observed to be unhealthy
+        represented as nanoseconds since the Unix epoch. Unhealthy is defined
+        as the component being in a state other than HEALTHY.";
+      oc-ext:telemetry-on-change;
+    }
+
+    leaf unhealthy-count {
+      type uint64;
+      description
+        "The number of status checks that have determined this component
+        to be in an unhealthy state.";
+      oc-ext:telemetry-on-change;
+    }
+  }
```

Comment:
I wonder, should this counter increase when we use Healthz.Get and Healthz.Check RPC? Or should it instead only increase when the system detects the component unhealthy by doing an automatic check?

For example, if a component_A is in unhealthy state, should the counter increase when we do multiple Healthz.Get check targeting this component while it stays in the unhealthy state?

Reply:
I think this should only be incremented when the device itself finds the component to be unhealthy, regardless of how many times a user triggers another check. We can clarify the leaf's description to be clearer if this is useful,

The justification for this is that it's probably not that useful to have a counter that says "you triggered 100 checks on this component and they are all unhealthy", but it is probably reasonable to say that "this component became unhealthy twice" (i.e., it transitioned between HEALTHY and UNHEALTHY twice), since this tells you that this is a new `UNHEALTHY` state, not the last one you have mitigated.

@marcushines - PTAL.

Reply:
Done :-)